crates/services/p2p/src/service.rs

```diff
@@ -1601,6 +1635,9 @@ pub mod tests {
             heartbeat_max_time_since_last,
             next_check_time: Instant::now(),
             heartbeat_peer_reputation_config: heartbeat_peer_reputation_config.clone(),
+            cached_view: Arc::new(CachedView::new(false)),
+            cache_reset_interval: Duration::from_secs(0),
```

Comment:
We could consider adding a test when the reset interval is >0 to check if it is respected.